tests/table.spec.ts

```diff
@@ -0,0 +1,139 @@
+import { test, expect, Page } from '@playwright/test';
+import { generateVarName } from './utils';
+
+// Run tests serially since they all use the same table
+test.describe.configure({ mode: 'serial' });
+
+let page: Page;
+
+test.beforeAll(async ({ browser }) => {
+  page = await browser.newPage();
+  await page.goto('http://localhost:4000/');
+});
+
+test.afterAll(async () => {
+  await page.close();
+});
+
+test('can open a simple table', async () => {
+  const consoleInput = page.locator('.console-input');
+  await consoleInput.click();
+
+  // Enter a command that creates a table with three columns, showing x/y/z
+  const tableName = generateVarName();
+  await page.keyboard.type(
+    `from deephaven import empty_table\n${tableName} = empty_table(100).update(["x=i", "y=Math.sin(i)", "z=Math.cos(i)"])`
+  );
+  await page.keyboard.press('Enter');
+
+  // Wait for the panel to show
+  await expect(page.locator('.iris-grid-panel')).toHaveCount(1);
+
+  // Wait until it's done loading
+  await expect(page.locator('.iris-grid-panel .loading-spinner')).toHaveCount(
+    0
+  );
+
+  // Model is loaded, need to make sure table data is also loaded
+  await expect(
+    page.locator('.iris-grid .iris-grid-loading-status')
+  ).toHaveCount(0);
+
+  // Now we should be able to check the snapshot
+  await expect(page.locator('.iris-grid-panel .iris-grid')).toHaveScreenshot();
+});
+
+test.describe('tests table operations', () => {
+  test.beforeEach(async () => {
+    const tableOperationsMenu = page.locator(
+      'data-testid=btn-iris-grid-settings-button-table'
+    );
+    await tableOperationsMenu.click();
+
+    // Wait for Table Options menu to show
+    await expect(page.locator('.table-sidebar')).toHaveCount(1);
+  });
+
+  test('can download table successfully', async () => {
+    // open Download CSV panel
+    await page.locator('data-testid=menu-item-Download CSV').click();
+
+    const downloadButton = page.locator(
+      'data-testid=btn-csv-exporter-download'
+    );
+    expect(downloadButton).toHaveCount(1);
+
+    // try renaming file before downloading
+    await page.locator('data-testid=input-csv-exporter-file-name').click();
+    await page.keyboard.press('Control+a');
+    await page.keyboard.type('sin-and-cos.csv');
+
+    downloadButton.click();
+
+    // Wait for download to complete
+    await expect(
+      page.locator('.progress.progress-bar-striped.progress-bar-animated')
+    ).toHaveCount(0);
+
+    await expect(
+      page.locator('.progress .progress-bar.bg-success')
+    ).toHaveCount(1);
+
+    // Check snapshot
+    await expect(
+      page.locator('.iris-grid .navigation-page')
+    ).toHaveScreenshot();
```

Comment:
I wouldn't bother with a screenshot here - just check that the progress bar is there, that it shows success, etc. is fine.

Screenshots/snapshots are mainly used for canvas elements and things like that.